sycl/include/sycl/ext/oneapi/experimental/sycl_complex.hpp

```diff
@@ -1003,6 +1003,448 @@ constexpr complex<float> operator""if(unsigned long long __im) {
+    for (std::size_t i = 0; i < NumElements; ++i) {                            \
+      rtn[i] = sycl::ext::oneapi::experimental::math_func(x[i], y[i]);         \
+    }                                                                          \
+    return rtn;                                                                \
+  }                                                                            \
+                                                                               \
+  template <typename T, std::size_t NumElements,                               \
+            typename = std::enable_if<is_genfloat<T>::value ||                 \
+                                      is_gencomplex<T>::value>>                \
+  _SYCL_EXT_CPLX_INLINE_VISIBILITY sycl::marray<rtn_type, NumElements>         \
+  math_func(const sycl::marray<arg_type1, NumElements> &x,                     \
+            const arg_type2 &y) {                                              \
+    sycl::marray<rtn_type, NumElements> rtn;                                   \
+    for (std::size_t i = 0; i < NumElements; ++i) {                            \
+      rtn[i] = sycl::ext::oneapi::experimental::math_func(x[i], y);            \
+    }                                                                          \
+    return rtn;                                                                \
+  }                                                                            \
+                                                                               \
+  template <typename T, std::size_t NumElements,                               \
+            typename = std::enable_if<is_genfloat<T>::value ||                 \
+                                      is_gencomplex<T>::value>>                \
+  _SYCL_EXT_CPLX_INLINE_VISIBILITY sycl::marray<rtn_type, NumElements>         \
+  math_func(const arg_type1 &x,                                                \
+            const sycl::marray<arg_type2, NumElements> &y) {                   \
+    sycl::marray<rtn_type, NumElements> rtn;                                   \
+    for (std::size_t i = 0; i < NumElements; ++i) {                            \
+      rtn[i] = math_func(x, y[i]);                                             \
+    }                                                                          \
+    return rtn;                                                                \
+  }
+
+MATH_OP_TWO_PARAM(pow, complex<T>, complex<T>, T);
+MATH_OP_TWO_PARAM(pow, complex<T>, complex<T>, complex<T>);
+MATH_OP_TWO_PARAM(pow, complex<T>, T, complex<T>);
+
+#undef MATH_OP_TWO_PARAM
+
+// Special definition as polar requires default argument
+
+template <typename T, std::size_t NumElements,
+          typename = std::enable_if<is_genfloat<T>::value>>
+_SYCL_EXT_CPLX_INLINE_VISIBILITY
+    sycl::marray<sycl::ext::oneapi::experimental::complex<T>, NumElements>
```

Comment:
I think I'd prefer using `std::enable_if` on the return type.

Reply:
> Is it just for this case?

No, generally.

> Why do you prefer it?

Because it doesn't require an artificial parameter that could technically be supplied by the caller (even though it's not supposed to be done).